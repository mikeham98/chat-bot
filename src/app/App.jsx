import React from 'react';
import ThemeContainer from "./containers/Theme/ThemeContainer";
import ChatBotLayout from "./layout/ChatBotLayout";
import ConversationPanelLayout from "./layout/ConversationPanelLayout";
import styles from '../themes/components/layout.scss';

export default class App extends React.Component {
    render() {
        return (
            <ThemeContainer>
                <div className={styles.chatBotContainer}>
                    <div className={styles.chatBotWrapper}>
                        <ConversationPanelLayout/>
                        <div className={styles.messageSection}>
                            <ChatBotLayout/>
                        </div>
                    </div>
                </div>
            </ThemeContainer>
        );
    }
}