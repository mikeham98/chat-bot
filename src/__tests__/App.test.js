import React from 'react';
import {shallow} from 'enzyme';
import App from '../app/App';

describe('App', () => {

    let shallowedWrapper;
    let props = {};
    const wrapper = () => {
        if (!shallowedWrapper) {
            shallowedWrapper = shallow(<App {...props}/>);
        }
        return shallowedWrapper;
    };

    beforeEach(() => {
        props = {};
        shallowedWrapper = undefined;
    });

    describe('render', () => {
        it('should return a ThemeContainer, ConversationPanelLayout and ChatBotLayout', () => {
            expect(wrapper()).toMatchSnapshot();
        });
    });
});