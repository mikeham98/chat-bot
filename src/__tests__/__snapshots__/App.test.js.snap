// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render should return a ThemeContainer, ConversationPanelLayout and ChatBotLayout 1`] = `
<Connect(ThemeContainer)>
  <div
    className="chatBotContainer"
  >
    <div
      className="chatBotWrapper"
    >
      <ConversationPanelLayout />
      <div
        className="messageSection"
      >
        <ChatBotLayout />
      </div>
    </div>
  </div>
</Connect(ThemeContainer)>
`;
